Use greater than or equal to in resample check

lboeman opened the pull request:
<!-- Thank you for your contribution! The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items. Feel free to remove checklist items that are not relevant to your change. -->

  - [x] Closes #723 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [ ] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

<!--
Brief description of the problem and proposed solution (if not already fully described in the issue linked to above): -->

lboeman commented on solarforecastarbiter/metrics/preprocessing.py:
```diff
@@ -396,7 +396,12 @@ def apply_flag(quality_flag):
         threshold = (
             quality_flag.resample_threshold_percentage / 100. * interval_ratio)
-        flagged = resampled_flags_count > threshold
+        # If threshold is 0, any points being flagged counts, but
+        # don't just throw away all data.
+        if threshold == 0:
+            flagged = resampled_flags_count > threshold
+        else:
+            flagged = resampled_flags_count >= threshold
```
@wholmgren What are your thoughts on handling a resample threshold of 0 this way?  If we don't do this, 0 just means "drop all the data".

wholmgren replied:
Ohh maybe that is why we chose > originally. I think that would be reasonable. But how would we document and communicate in the user interface?

lboeman replied:
We could either add it to the help text pop up, or add a note below the resample threshold percentage input. I think adding the note below is a better solution, just so that it's visible to users who already understand how these filters work.

wholmgren replied:
Sounds good.

wholmgren commented on solarforecastarbiter/metrics/tests/test_preprocessing.py:
```diff
@@ -506,8 +506,8 @@ def test_align_prob_constant_value(
             # only first all-nan interval is discarded
             pd.Series([7.0, 10.5, 14.5], index=FOUR_HOUR_SERIES.index[1:]),
             (('ISNAN', 5), ('TOTAL DISCARD BEFORE RESAMPLE', 5),
-             ('NIGHTTIME OR USER FLAGGED OR ISNAN', 0, False),
-             ('TOTAL DISCARD AFTER RESAMPLE', 0, False))
```
Maybe it has to do with the fact that the four 15 minute obs in the hour were all nan values instead of all some flag. But I still don't see how that works out.

lboeman replied:
I think this is found in `_calc_discard_before_resample` here: https://github.com/SolarArbiter/solarforecastarbiter-core/blob/23122548ec968ea4ec59f4877cc65f6e23994d4e/solarforecastarbiter/metrics/preprocessing.py#L314-L330

ISNAN is *always* discarded before resample, so we would expect that first interval to disappear in the output. However, the `to_discard_after_resample` is calculated based on the index of the `to_discard_before_resample` series https://github.com/SolarArbiter/solarforecastarbiter-core/blob/23122548ec968ea4ec59f4877cc65f6e23994d4e/solarforecastarbiter/metrics/preprocessing.py#L374-L375
which will then detect that it should discard that interval, because the ISNAN flag was present for 100% of the data.

wholmgren replied:
Thanks, now I get it. We should have expected the interval to be dropped because it was all NAN. And the fact that it was dropped while `_calc_discard_after_resample` returned flags with `count=0` should have been a warning that something was wrong.

lboeman commented:
I added a few more tests to explicitly check the changes made here and reverted the changed test that you commented on. It's hanging up on the dev api tests still but new tests are passing.

wholmgren reviewed (approved):
Those new tests are very helpful. Thanks.
